src/models/model.cpp

```diff
@@ -369,8 +377,121 @@ DeviceInterface* SetProviderSessionOptions(OrtSessionOptions& session_options,
         session_options.AddConfigEntry("session.inter_op.allow_spinning", "0");
         session_options.AddConfigEntry("session.intra_op.allow_spinning", "0");
       } else if (provider_options.name == "NvTensorRtRtx") {
-        // After setting the NvTensorRtRtx provider in Onnxruntime, GenAI will then treat it as the cuda device.
+        // Always enable CUDA graph
         session_options.AddConfigEntry("ep.nvtensorrtrtxexecutionprovider.nv_cuda_graph_enable", "1");
+
+        // Check for multi-profile option
+        if (const auto opt_it = std::find_if(provider_options.options.begin(), provider_options.options.end(),
+                                             [](const auto& pair) { return pair.first == "nv_multi_profile_enable"; });
+            opt_it != provider_options.options.end() && opt_it->second == "1") {
+
+          // Get model parameters from decoder config
+          const int num_layers = model_config.decoder.num_hidden_layers;
+          const int num_kv_heads = model_config.decoder.num_key_value_heads;
+          const int head_dim = model_config.decoder.head_size;
+
+          // Get max context length from config
+          const int max_context_len = model_config.context_length;
+          const int opt_context_len = 512;  // Optimal is half of max, capped at 512
+          const int min_seq_len = 1;
+
+          // Extract KV cache name patterns from decoder config
+          std::string past_key_pattern = model_config.decoder.inputs.past_key_names;
+          std::string past_value_pattern = model_config.decoder.inputs.past_value_names;
+
+          // If patterns are empty, use defaults
+          if (past_key_pattern.empty()) past_key_pattern = "past_key_values.%d.key";
+          if (past_value_pattern.empty()) past_value_pattern = "past_key_values.%d.value";
+
+          std::ostringstream min_shapes, opt_shapes, max_shapes;
+
+          // MIN SHAPES (context phase and first token generation)
+          min_shapes << "input_ids:1x" << min_seq_len << ",attention_mask:1x" << min_seq_len 
```

Comment:
Can we use the config's default string names for all of the strings that are created here?

https://github.com/microsoft/onnxruntime-genai/blob/375f1b61aab14c3931cc732b3c4c1fc5b86eafba/src/config.h#L16-L18

For example:

```suggestion
          min_shapes << Config::InputIdsName << ":1x" << min_seq_len << "," << Config::AttentionMaskName << ":1x" << min_seq_len 
```